src/librustc/middle/resolve_lifetime.rs

```diff
@@ -917,6 +917,16 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                 }
             }
             self.insert_lifetime(lifetime_ref, def);
+        } else if lifetime_ref.is_static() {
+            // The only way we end up here is if there is a static lifetime bound (due to early
+            // return on `visit_lifetime()`)
+            self.insert_lifetime(lifetime_ref, Region::Static);
+            let full_span = mk_sp(context_ref.span.lo, lifetime_ref.span.hi);
```

Comment:
This would become `context_ref.unwrap().span.lo`, enforcing the invariant described in the above comment.